### PR TITLE
Extract intent and entities using luis bot api - first stage

### PR DESCRIPTION
**Pull request by sreeragkiyyath**

### CODE REVIEW

**Comment by sreeragkiyyath** [comment]
Please add the key while using luis_setter.py

**Comment by amartyaamp** [comment]
Can you post the test code (with this pull request) which creates LuisBotSetter object and calls getTokens and show the output ?
Preferably in a separate test.py .

**Comment by amartyaamp** [comment]
@sreeragkiyyath  Can you test with the string - _" send a mail to abc@gmail.com "_ ?  
On my side it did not return any intent.

**Comment by sreeragkiyyath** [comment]
Try again, Please list out unsuccessful queries here

File: src/luis_setter.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      intentFromQuery = entityIntent.get('topScoringIntent')['intent']
+      entitiesFromQuery = entityIntent.get('entities')
+      returnString = "intent:" + intentFromQuery +",entity:("
+      for index, entities in enumerate(entitiesFromQuery): 
+        if 'resolution' in entities.keys():
+          mappingValue = entities.get('resolution')['values'][0].get('value')
+        
+        returnString = returnString + str(index)  + ":" + entities['entity'] + "(" + mappingValue + "),"
+      returnString = returnString + ")"      
+      print(returnString)
```
Comment (amartyaamp) [inline src/luis_setter.py]: Function doesn't return anything!

File: src/luis_setter.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+    params['q'] = queryFromUser
+    try:
+      r = requests.get ('https://southeastasia.api.cognitive.microsoft.com/luis/v2.0/apps/24c19c39-9242-4cea-ab89-4213e1f146cc', headers = self.headers, params = params)
+      #print (r.json())
+	    #https://westus.api.cognitive.microsoft.com/api/v2.0/apps/24c19c39-9242-4cea-ab89-4213e1f146cc?subscription-key=ee8f4937e8c94ef9b79ce873d9cce89d&verbose=true&timezoneOffset=330&q=	
+      entityIntent = r.json() 
+      intentFromQuery = entityIntent.get('topScoringIntent')['intent']
+      entitiesFromQuery = entityIntent.get('entities')
+      returnString = "intent:" + intentFromQuery +",entity:("
```
Comment (amartyaamp) [inline src/luis_setter.py]: Do not return a unformatted string - **_returnString_**. This causes difficulty in parsing on the other side. Form a dictionary of the result and return the json dump. Using json.dumps(dictionary)

Comment (amartyaamp) [inline src/luis_setter.py]: Try forming a dictionary such that -  

1. **_tokens['intent']_** gives me the top scoring intent.
2. **_tokens['entities']_** gives me a list of entities and their type in the statement.

Create a **_json_tokens = json.dumps(tokens)_** and return this

**Review by amartyaamp — CHANGES_REQUESTED** [review]
Some changes that are required before we can plugin.

**Comment by amartyaamp** [comment]
@sreeragkiyyath  
Till now following are the unsuccessful queries - 
1.  _**Do i have any mail from abc@xyz.com ?**_ ---- > **_intent:Calendar.Find,entity:(0:abc@xyz.com(),)_**
   This is not related to Calendar.
2. Also, can we return the type of entity here?

**Pull request merged**